src/scenic/core/object_types.py

```diff
@@ -3,666 +3,1760 @@
-		return (
-			self.relativePosition(Vector(hw, hl)),
-			self.relativePosition(Vector(-hw, hl)),
-			self.relativePosition(Vector(-hw, -hl)),
-			self.relativePosition(Vector(hw, -hl))
-		)
-
-	def show(self, workspace, plt, highlight=False):
-		if needsSampling(self):
-			raise RuntimeError('tried to show() symbolic Object')
-		pos = self.position
-		spos = workspace.scenicToSchematicCoords(pos)
-
-		if highlight:
-			# Circle around object
-			rad = 1.5 * max(self.width, self.length)
-			c = plt.Circle(spos, rad, color='g', fill=False)
-			plt.gca().add_artist(c)
-			# View cone
-			ha = self.viewAngle / 2.0
-			camera = self.position.offsetRotated(self.heading, self.cameraOffset)
-			cpos = workspace.scenicToSchematicCoords(camera)
-			for angle in (-ha, ha):
-				p = camera.offsetRadially(20, self.heading + angle)
-				edge = [cpos, workspace.scenicToSchematicCoords(p)]
-				x, y = zip(*edge)
-				plt.plot(x, y, 'b:')
-
-		corners = [workspace.scenicToSchematicCoords(corner) for corner in self.corners]
-		x, y = zip(*corners)
-		color = self.color if hasattr(self, 'color') else (1, 0, 0)
-		plt.fill(x, y, color=color)
-
-		frontMid = averageVectors(corners[0], corners[1])
-		baseTriangle = [frontMid, corners[2], corners[3]]
-		triangle = [averageVectors(p, spos, weight=0.5) for p in baseTriangle]
-		x, y = zip(*triangle)
-		plt.fill(x, y, "w")
-		plt.plot(x + (x[0],), y + (y[0],), color="k", linewidth=1)
+class Object(OrientedPoint):
+    """The Scenic class ``Object``.
+
+    This is the default base class for Scenic classes.
+
+    Properties:
+        width (float): Width of the object, i.e. extent along its X axis.
+          Default value of 1 inherited from the object's :prop:`shape`.
+        length (float): Length of the object, i.e. extent along its Y axis.
+          Default value of 1 inherited from the object's :prop:`shape`.
+        height (float): Height of the object, i.e. extent along its Z axis.
+          Default value of 1 inherited from the object's :prop:`shape`.
+        shape (`Shape`): The shape of the object, which must be a `Shape` class.
+          The default shape is a box, with default unit dimensions.
+        allowCollisions (bool): Whether the object is allowed to intersect
+          other objects. Default value ``False``.
+        regionContainedIn (`Region` or ``None``): A `Region` the object is
+          required to be contained in. If ``None``, the object need only be
+          contained in the scenario's workspace.
+        baseOffset (`Vector`): An offset from the :prop:`position` of the Object
+          to the base of the object, used by the ``on`` specifier. Default value
+          is (0, 0, -self.height/2), placing the base of the Object at the bottom
+          center of the Object's bounding box.
+        contactTolerance (float): The distance away an object should be placed
+          from a point to be on the point, used by the ``on`` specifier. Default
+          value 1e-4.
```

Comment:
This description isn't the same as that given in `Point`. Should also mention that this default value overrides the one from `Point`.